mlrun/execution.py

```diff
@@ -497,27 +497,34 @@ def _set_input(self, key, url=""):
             url = key
         if self.in_path and is_relative_path(url):
             url = os.path.join(self._in_path, url)
-        obj = self._data_stores.object(
-            url,
-            key,
-            project=self._project,
-            allow_empty_resources=self._allow_empty_resources,
-        )
-        self._inputs[key] = obj
-        return obj
+        self._inputs[key] = url
 
     def get_input(self, key: str, url: str = ""):
-        """get an input :py:class:`~mlrun.DataItem` object, data objects have methods such as
-        .get(), .download(), .url, .. to access the actual data
+        """
+        Get an input :py:class:`~mlrun.DataItem` object,
+        data objects have methods such as .get(), .download(), .url, .. to access the actual data.
+        Requires access to the data store secrets if configured.
 
-        example::
+        Example::
 
             data = context.get_input("my_data").get()
+
+        :param key:  The key name for the input url entry.
+        :param url:  The url of the input data (file, stream, ..) - optional, saved in the inputs dictionary
+                     if the key is not already present.
+
+        :return:     :py:class:`~mlrun.datastore.base.DataItem` object
         """
         if key not in self._inputs:
-            return self._set_input(key, url)
-        else:
-            return self._inputs[key]
+            self._set_input(key, url)
+
+        url = self._inputs[key]
+        return self._data_stores.object(
+            url,
+            key,
+            project=self._project,
```

Comment:
why we save the input in the inputs list if the input wasn't stored purposely inside the context? what we gain from that?

Reply:
the context getters are used as "get or set" logic and it's true for all of them.
can debate on if it's correct or not but we should make sure all the getters are aligned to avoid confusion.

Reply:
but why? does it mean that I can ask for an input that the function didn't had defined as part of the handler, and now it would be added to the inputs context?